### ISSUE TITLE
Sub-form response that allows multiple responses

### ISSUE THREAD

**Issue by zzmgck** [nextcloud/forms#1728]
**Nextcloud (please complete the following information):**
- Nextcloud-Version: 26.0.x
- Forms-Version: 3.3.x

**Is your feature request related to a problem? Please describe.**
I need the ability to construct a response that I would describe as a sub-form that allows the user to add an arbitrary number of responses to a list.  For example, a travel form that allows the user to make an itinerary of stops

**Describe the solution you'd like**
The feature would allow the user to design a form and then use that form as sub-form.  In the travel example, I would make a travel stop form that has an arrival date, departure date, city, country.

The main form would have a list element that would allow the user to add an arbitrary number of stops to the itinerary element of the travel form.

Ideally, the form designer could designate an element in the sub-form as the sort element that would sort the responses within the list.

**Describe alternatives you've considered**
A large text block; however, there is no way to enforce formatting and field validation (e.g. valid dates).

**Comment by Chartman123** [comment]
I think this is rather an edge use case, so we'll probably not be implementing this any time soon.

**Comment by Chartman123** [comment]
@jotoeri @susnux What do you think about a question type "sub-form"?? The more I think about it, the more I come to the conclusion that this would kind of abuse the original form for a use case that probably isn't needed very often and could be better suited by some other application...

**Comment by susnux** [comment]
I agree with @Chartman123 currently we follow the KIS principle, so keeping it simple but that seems to be a corner case that would introduce much overhead and complexity.

**Comment by zzmgck** [comment]
As the submitter, I do think keeping the present Forms simple is a good decision. The idea of a new application that has the additional complexity is a good one.  I have not looked at the source, but my initial thought would be to base a new application that attempts to use the Forms application as a component.  Does the current architecture lend itself to reuse?
